### ISSUE TITLE
Message Status - Microsoft.Azure.ServiceBus

### ISSUE THREAD

**Issue by arunkumaresan** [Azure/azure-service-bus-dotnet#414]
## Actual Behavior
1. Message Status property not available

## Expected Behavior
1. Message Status property required
2. 

## Versions  
- Microsoft.Azure.ServiceBus -> class Message
- NuGet package version or commit ID: 2.0.0.0

**Comment by binzywu** [comment]
@arunkumaresan Sorry for the late reply. Do you mean MessageState of BrokeredMessage in the old client library?

**Comment by arunkumaresan** [comment]
@binzywu Yes, the message State is represented in Old Library, but it is not reflected in New library

**Comment by SeanFeldman** [comment]
[`BrokeredMessage.State`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.brokeredmessage.state?view=azure-dotnet#Microsoft_ServiceBus_Messaging_BrokeredMessage_State). Sounds like it should be on the SystemProperties collection as information coming from the broker

**Comment by rudym** [comment]
Would be really nice to have that information, currently, we have to distinct deferred messages from a batch of messages using hacky filtering 

```
m.SystemProperties.DeliveryCount > 1 &&
					m.SystemProperties.LockedUntilUtc.Kind != DateTimeKind.Unspecified
```

and I would appreciate if anyone can point us to a better approach in meanwhile.

**Comment by stridsbr** [comment]
I also need this information on the message.systemproperties. @rudym did you find a better solution?
